encodings/dict/src/builders/mod.rs

```diff
@@ -29,7 +30,8 @@ pub fn dict_encode_max_sized(array: &dyn Array, max_dict_bytes: usize) -> Vortex
     } else {
         vortex_bail!("Can only encode primitive or varbin/view arrays")
     };
-    let codes = dict_builder.encode(array)?;
+    let codes = downscale_integer_array(dict_builder.encode(array)?)?;
```

Comment:
you also don't always go through this code for dict. There's logic in btrblocks compressor to dict encode from fixed dictionary

Reply:
it has its own downscaling logic, it uses the length of the values array to determine the type